testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/DeltaQueryExecutor.java

```diff
@@ -0,0 +1,139 @@
+        extends JdbcQueryExecutor
+{
+    private static final DeltaConnectionsPool CONNECTIONS_POOL = new DeltaConnectionsPool();
+
+    public DeltaQueryExecutor(TestContext testContext)
+    {
+        super(testContext.getDependency(JdbcConnectivityParamsState.class, "delta"), CONNECTIONS_POOL, testContext);
+    }
+
+    private static class DeltaConnectionsPool
+            extends JdbcConnectionsPool
+    {
+        private final Cache<String, Driver> drivers = buildNonEvictableCache(CacheBuilder.newBuilder().maximumSize(2));
+        private Connection connection;
+
+        @Override
+        public Connection connectionFor(JdbcConnectivityParamsState jdbcParamsState)
+                throws SQLException
+        {
+            if (connection == null || connection.isClosed()) {
+                Properties properties = new Properties();
+                properties.put("user", jdbcParamsState.user);
+                properties.put("password", jdbcParamsState.password);
+                connection = uncheckedCacheGet(drivers, jdbcParamsState.driverClass, () -> getDatabaseDriver(jdbcParamsState.driverClass))
+                        .connect(jdbcParamsState.url, properties);
+            }
+
+            if (connection == null) {
+                // this should never happen, `javax.sql.DataSource#getConnection()` should not return null
+                throw new IllegalStateException("No connection was created for: " + jdbcParamsState.getName());
+            }
+            if (!jdbcParamsState.prepareStatements.isEmpty()) {
+                try (Statement statement = connection.createStatement()) {
+                    for (String query : jdbcParamsState.prepareStatements) {
+                        try {
+                            statement.execute(query);
+                        }
+                        catch (SQLException e) {
+                            throw new SQLException("Preparatory statement failed: " + query, e);
+                        }
+                    }
+                }
+            }
+            return connection;
+        }
+    }
+
+    private static Driver getDatabaseDriver(String driverClassName)
+    {
+        // TODO Add support for maven coordinate in tempto
+        ArtifactResolver resolver = new ArtifactResolver(USER_LOCAL_REPO, ImmutableList.of(MAVEN_CENTRAL_URI));
+        List<URL> classPath = resolver.resolveArtifacts(getArtifact(driverClassName)).stream()
+                .map(artifact -> {
+                    try {
+                        return artifact.getFile().toURI().toURL();
+                    }
+                    catch (MalformedURLException e) {
+                        throw new RuntimeException(e);
+                    }
+                })
+                .collect(toImmutableList());
+
+        checkArgument(!classPath.isEmpty(), "classPath must not be empty");
+        URLClassLoader classLoader = new URLClassLoader(classPath.toArray(URL[]::new), getPlatformClassLoader());
```

Comment:
My IDE shows a warning (`'URLClassLoader' used without 'try'-with-resources statement`). It should be suppressed.